app/assets/javascripts/angular/components/surveyOverviewQuestionsTab/surveyOverviewQuestions.component.js

```diff
@@ -52,21 +61,162 @@ function surveyOverviewQuestionsController($uibModal, surveyService) {
             name: 'Dropdown (Choose one)',
             canAdd: true,
         },
-        { kind: 'DateField', style: 'date_field', name: 'Date' },
+        {
+            kind: 'DateField',
+            style: 'date_field',
+            name: 'Date',
+        },
     ];
+    this.people = [];
 
-    this.getQuestionType = (kind, style) => {
-        return _.find(this.questionTypes, { kind: kind, style: style });
+    const loadSurveyData = async () => {
+        let q = await surveyService.getSurveyQuestions(this.survey.id);
+        this.people = await getPeople(q, this.survey.organization_id);
+        rebuildQuestions(q);
+        $scope.$apply();
     };
 
-    this.sortableOptions = {
-        handle: '.sort',
+    const getPeople = async (questions, organizationId) => {
+        let list = questions.reduce((a1, q) => {
+            return (
+                a1 +
+                q.question_rules.reduce((a2, r) => {
+                    if (!r.people_ids) {
+                        return a2;
+                    }
+
+                    return a2 + r.people_ids + ',';
+                }, '')
+            );
+        }, '');
+
+        let peopleIds = [...new Set(list.split(','))];
+
+        let r = await httpProxy.get(
+            '/people',
+            {
+                'filters[ids]': peopleIds.join(','),
+                'filters[organization_ids]': this.survey.organization_id,
+                'fields[people]':
+                    'first_name,gender,last_name,primary_email_address,id',
+            },
+            {
+                errorMessage: 'error.messages.surveys.loadQuestions',
+            },
+        );
+
+        return r.data;
+    };
+
+    const rebuildQuestions = questions => {
+        this.surveyQuestions = questions.map(q => buildQuestion(q));
+    };
+
+    const rebuildQuestion = question => {
+        let q = buildQuestion(question);
+        let index = this.surveyQuestions.indexOf(question);
+        this.surveyQuestions[index] = q;
+    };
+
+    const buildQuestion = question => {
+        let q = question;
+        let a = question.content ? question.content.split('\n') : [];
+        q.question_answers = a;
```

Comment:
Could define the whole thing at once instead over a couple lines.
```suggestion
        const q = {
          ...question,
          question_answers: a,
        }
```

Reply:
yup, actually just refactored a bit more, dont even need q, and no longer mutation :)  Yah the ... is beautful, arrays and objects...